Bindings/Java/Matlab/Utilities/osimC3Dconverter.m

```diff
@@ -0,0 +1,278 @@
+filepath = p.Results.filepath;
+axis = p.Results.axis;
+value = p.Results.value;
+
+%% Check for file path
+if isempty(filepath)
+        [filein, path] = uigetfile({'*.c3d','C3D file'}, 'Select C3D data file(s)...', 'MultiSelect', 'on');
+else
+    % If the input path to file is wrong, return exception
+    if exist(filepath,'file') == 0
+            error('file does not exist')
+    else    
+        % If the input path is local (called from current folder), you will
+        % need to set the full path. 
+        [path, filename, ext] = fileparts(filepath);
+        if isempty(path)
+            path = cd;
+        end
+        filein = [filename ext];
+    end
+end
+
+if iscell(filein)
+    nFiles = length(filein);
+    multipleFiles = 1;
+else
+    nFiles = 1;
+    multipleFiles = 0;
+end
+    
+for iFile = 1 : nFiles
+    
+    if multipleFiles == 1
+        % get the file name (with extension)
+        filename = filein{iFile};
+    else 
+        filename = filein;
+    end    
+        
+     % get the fullpath to the file 
+    filepath = fullfile(path,filename);
+    % get the file name (without the extension)
+    [path, filename, ext] = fileparts(filepath);
+
+    %% import java libraries
+    import org.opensim.modeling.*
+
+    %% Use a c3dAdapter to turn read a c3d file
+    adapter = C3DFileAdapter();
+    tables = adapter.read(filepath);
+
+    %% get the marker and force data into OpenSim tables
+    markers = tables.get('markers');
+    forces = tables.get('forces');
+
+    %% Rotate marker and force data
+    if isempty(axis)
+        display('Either no axis or value for rotation has been been set; no rotations will be applied')
+        forces_rot = forces.clone();
+        markers_rot = markers.clone();
+    else
+        display(['Rotating Marker and Force data ' num2str(value) ' about the ' axis ' axis']); 
+        markers_rot = osimRotateTableData(markers, axis, value);
+        forces_rot = osimRotateTableData(forces, axis, value);
+    end
+    
+    %% Print the rotated markers to trc file
+    TRCFileAdapter().write(markers_rot,fullfile(path,[filename '.trc']));
+    display([[filename '.trc'] ' written to dir: ' path]);
+    %% Print the force data as a Vec3 sto file and a flattened doubles sto file
+    % make postfix string vector for naming colomns
+
+    %% flatten the Vec3 table to a table of doubles. 
+    postfix = StdVectorString(); postfix.add('_x');postfix.add('_y');postfix.add('_z');
+    forces_flattened = forces_rot.flatten(postfix);
+
+    % update the column names to have v,p,m n them.
+
+    labels = forces_flattened.getColumnLabels();
+    newlabels = forces_flattened.getColumnLabels();
+
+    for i = 0 : labels.size() - 1
+
+        label = char(labels.get(i));
+
+        if i < 9
+            n = '1';
+        elseif  8 < i < 18
+            n = '2';
+        elseif 17 < i < 27
+           n = '3';
+        end
+
+        if ~isempty(strfind(label,'f'))
+            s = ['ground_force_' n '_v'];
```

Comment:
Hmmm can you send me the file you tried without `ground_force`? 
Just to confirm, the file attached didn't preview for you?